File: attentive-android-sdk/src/main/java/com/attentive/androidsdk/internal/network/Metadata.java
package com.attentive.androidsdk.internal.network;

import com.attentive.androidsdk.UserIdentifiers;
import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class Metadata {
    private String phone;
    private String email;
    private String source = "msdk";

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public void enrichWithIdentifiers(UserIdentifiers userIdentifiers) {
        if (this.getEmail() == null && userIdentifiers.getEmail() != null) {
            this.setEmail(userIdentifiers.getEmail());
        }
        if (this.getPhone() == null && userIdentifiers.getPhone() != null) {
            this.setPhone(userIdentifiers.getPhone());
        }
    }
}

File: attentive-android-sdk/src/main/java/com/attentive/androidsdk/internal/network/Metadata.kt
package com.attentive.androidsdk.internal.network

import com.attentive.androidsdk.UserIdentifiers
import kotlinx.serialization.EncodeDefault
import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.Polymorphic
import kotlinx.serialization.Serializable

@OptIn(ExperimentalSerializationApi::class)
@Serializable
@Polymorphic
open class Metadata {
    var phone: String? = null
    var email: String? = null

    @EncodeDefault
    var source: String = "msdk"

    fun enrichWithIdentifiers(userIdentifiers: UserIdentifiers) {
        if (this.email == null && userIdentifiers.email != null) {
            this.email = userIdentifiers.email
        }
        if (this.phone == null && userIdentifiers.phone != null) {
            this.phone = userIdentifiers.phone
        }
    }
}    var currency: String? = null
